system-contracts/test/L2EthToken.spec.ts

```diff
@@ -0,0 +1,222 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import type { Wallet } from "zksync-web3";
+import type { L2EthToken } from "../typechain";
+import { IMailboxFactory } from "../typechain/IMailboxFactory";
+import { L2EthTokenFactory } from "../typechain";
+import { deployContractOnAddress, getWallets, provider } from "./shared/utils";
+import type { BigNumber } from "ethers";
+import { TEST_BOOTLOADER_FORMAL_ADDRESS, TEST_ETH_TOKEN_SYSTEM_CONTRACT_ADDRESS } from "./shared/constants";
+import { prepareEnvironment, setResult } from "./shared/mocks";
+import { randomBytes } from "crypto";
+
+describe("L2EthToken tests", () => {
+  const richWallet = getWallets()[0];
+  let wallets: Array<Wallet>;
+  let l2EthToken: L2EthToken;
+  let bootloaderAccount: ethers.Signer;
+
+  before(async () => {
+    await prepareEnvironment();
+    await deployContractOnAddress(TEST_ETH_TOKEN_SYSTEM_CONTRACT_ADDRESS, "L2EthToken");
+    l2EthToken = L2EthTokenFactory.connect(TEST_ETH_TOKEN_SYSTEM_CONTRACT_ADDRESS, richWallet);
+    bootloaderAccount = await ethers.getImpersonatedSigner(TEST_BOOTLOADER_FORMAL_ADDRESS);
```

Comment:
Can you stop impersonating in the `after`? Like here: https://github.com/matter-labs/era-contracts/blob/dev/system-contracts/test/Compressor.spec.ts#L37. It's not important for now, but can be in the future in case some tests will depend on it.